File: capacitor-codemirror/android/src/main/java/com/notes/plugins/codemirror/CodemirrorPlugin.java
package com.notes.plugins.codemirror;

import com.getcapacitor.JSObject;
import com.getcapacitor.Plugin;
import com.getcapacitor.PluginCall;
import com.getcapacitor.PluginMethod;
import com.getcapacitor.annotation.CapacitorPlugin;
import android.content.Intent;

@CapacitorPlugin(name = "Codemirror")
public class CodemirrorPlugin extends Plugin {

    private Codemirror implementation = new Codemirror();

    @PluginMethod
    public void echo(PluginCall call) {
        String value = call.getString("value");

        JSObject ret = new JSObject();
        ret.put("value", implementation.echo(value));
        call.resolve(ret);
    }

    @PluginMethod
    public void start(PluginCall call) {
        String path = call.getString("path");
        if (path == null) {
            call.reject("Path is required");
            return;
        }

        String title = call.getString("title", "Editor");
        String theme = call.getString("theme", "dark");

        Intent intent = new Intent(getContext(), CodemirrorActivity.class);
        intent.putExtra("path", path);
        intent.putExtra("title", title);
        intent.putExtra("theme", theme);
        
        // We start the activity. Note: if you want to get result back, use startActivityForResult
        getActivity().startActivity(intent);
        
        call.resolve();
    }
}

File: capacitor-codemirror/android/src/main/java/com/notes/plugins/codemirror/CodemirrorPlugin.kt
package com.notes.plugins.codemirror

import android.content.Intent
import com.getcapacitor.JSObject
import com.getcapacitor.Plugin
import com.getcapacitor.PluginCall
import com.getcapacitor.PluginMethod
import com.getcapacitor.annotation.CapacitorPlugin

@CapacitorPlugin(name = "Codemirror")
class CodemirrorPlugin : Plugin() {
    private val implementation = Codemirror()

    @PluginMethod
    fun echo(call: PluginCall) {
        val value = call.getString("value")
        val ret = JSObject()
        ret.put("value", implementation.echo(value ?: ""))
        call.resolve(ret)
    }

    @PluginMethod
    fun start(call: PluginCall) {
        val path = call.getString("path")
        if (path == null) {
            call.reject("Path is required")
            return
        }
        val title = call.getString("title", "Editor")
        val theme = call.getString("theme", "dark")
        val fontSize = call.getInt("fontSize", 16)
        val showLineNumbers = call.getBoolean("showLineNumbers", true)
        val autoSave = call.getBoolean("autoSave", true)
        val language = call.getString("language", "Plain Text")

        val intent = Intent(context, CodemirrorActivity::class.java)
        intent.putExtra("path", path)
        intent.putExtra("title", title)
        intent.putExtra("theme", theme)
        intent.putExtra("fontSize", fontSize)
        intent.putExtra("showLineNumbers", showLineNumbers)
        intent.putExtra("autoSave", autoSave)
        intent.putExtra("language", language)

        // We start the activity. Note: if you want to get result back, use startActivityForResult
        activity.startActivity(intent)
        call.resolve()
    }
}
